src/extension/background-script/connectors/alby.ts

```diff
@@ -0,0 +1,620 @@
+        type: string;
+        value: number;
+        metadata: string;
+      }[]
+    >("GET", "/invoices/incoming", undefined);
+
+    const invoices: ConnectorInvoice[] = incomingData
+      .map(
+        (invoice, index): ConnectorInvoice => ({
+          custom_records: invoice.custom_records,
+          id: `${invoice.payment_request}-${index}`,
+          memo: invoice.memo,
+          preimage: "", // lndhub doesn't support preimage (yet)
+          settled: invoice.settled,
+          settleDate: invoice.settled_at,
+          totalAmount: `${invoice.amount}`,
+          type: "incoming",
+        })
+      )
+      .concat(
+        outgoingData.map(
+          (invoice, index): ConnectorInvoice => ({
+            custom_records: invoice.custom_records,
+            id: `${invoice.payment_request}-${index}`,
+            memo: invoice.memo,
+            preimage: "", // lndhub doesn't support preimage (yet)
+            settled: invoice.ispaid,
+            settleDate: invoice.settled_at,
+            totalAmount: `${invoice.amount}`,
+            type: "outgoing",
+          })
+        )
+      )
+      .sort((a, b) => {
+        return b.settleDate - a.settleDate;
+      });
+    return {
+      data: {
+        invoices,
+      },
+    };
+  }
+
+  async getInfo(): Promise<GetInfoResponse> {
+    const alias = "🐝 getalby.com";
+    return {
+      data: {
+        alias,
+      },
+    };
+  }
+
+  async getBalance(): Promise<GetBalanceResponse> {
+    const { balance } = await this.request<{ balance: number }>(
+      "GET",
+      "/balance",
+      undefined
+    );
+    return {
+      data: {
+        balance: balance,
+      },
+    };
+  }
+
+  async sendPayment(args: SendPaymentArgs): Promise<SendPaymentResponse> {
+    const data = await this.request<{
+      error?: string;
+      message: string;
+      payment_error?: string;
+      payment_hash:
+        | {
+            type: string;
+            data: ArrayBuffer;
+          }
+        | string;
+      payment_preimage:
+        | {
+            type: string;
+            data: ArrayBuffer;
+          }
+        | string;
+      payment_route?: { total_amt: number; total_fees: number };
+    }>("POST", "/payments/bolt11", {
+      invoice: args.paymentRequest,
+    });
+    if (data.error) {
+      throw new Error(data.message);
+    }
+    if (data.payment_error) {
+      throw new Error(data.payment_error);
+    }
+    if (
+      typeof data.payment_hash === "object" &&
+      data.payment_hash.type === "Buffer"
+    ) {
+      data.payment_hash = Buffer.from(data.payment_hash.data).toString("hex");
+    }
+    if (
+      typeof data.payment_preimage === "object" &&
+      data.payment_preimage.type === "Buffer"
+    ) {
+      data.payment_preimage = Buffer.from(data.payment_preimage.data).toString(
+        "hex"
+      );
+    }
+
+    // HACK!
+    // some Lnbits extension that implement the LNDHub API do not return the route information.
```

Comment:
why is this needed? lnbits? lndhub? this is the new Alby oauth connector, it should use the Alby API via the alby js sdk?